File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1, num = nums[0];
        for(int i=1; i<nums.length; i++) {
            if(num == nums[i])
                count = count + 1;
            else 
                count = count - 1;
            
            if(count == 0) {
                num = nums[i];
                count = 1;
            }
        }

        return num;
    }
}

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let majority = nums[0], count = 1;

    for(let i = 1; i<nums.length; i++){
        if(majority == nums[i]){
            count ++;
        } else {
            count --
        }
        if(count == 0){
            majority = nums[i];
            count = 1;
        }
    }

    return majority;
};